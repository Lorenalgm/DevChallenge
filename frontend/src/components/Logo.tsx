import { Link } from "react-router-dom";

export function Logo() {
  return (
    <Link to="/" className="flex gap-4 items-center">
      <div className="flex gap-2 items-center justify-center bg-yellow-400 w-8 h-8 rounded-sm">
        <p className="text-lg font-bold text-black">D</p>
      </div>
      <h1 className="text-lg font-bold">
        <span className="text-yellow-400">Dev</span>Challenge
      </h1>
    </Link>
  );
}
