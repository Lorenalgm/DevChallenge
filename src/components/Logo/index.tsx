import { Heading, Box, HeadingProps } from '@chakra-ui/react'

const Logo = ({ ...props }: HeadingProps) => (
  <Heading textDecoration="none" {...props}>
    Dev
    <Box as="span" color="primary.400">
      Challenge
    </Box>
  </Heading>
)

export default Logo
