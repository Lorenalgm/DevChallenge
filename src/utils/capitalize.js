export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
