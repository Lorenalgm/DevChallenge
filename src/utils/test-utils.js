import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
