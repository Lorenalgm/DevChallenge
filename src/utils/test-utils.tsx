import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
