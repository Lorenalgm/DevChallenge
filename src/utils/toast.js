import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
