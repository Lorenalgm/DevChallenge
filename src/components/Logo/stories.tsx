import { Story, Meta } from '@storybook/react/types-6-0'

import Logo from '.'

export default {
  title: 'Logo',
  component: Logo
} as Meta

export const Default: Story = () => <Logo />
