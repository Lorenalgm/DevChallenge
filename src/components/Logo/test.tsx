import { render, screen } from 'utils/test-utils'

import Logo from '.'

describe('<Logo />', () => {
  it('should render the Logo as heading', () => {
    render(<Logo />)

    expect(
      screen.getByRole('heading', { name: /Dev Challenge/i, level: 2 })
    ).toBeInTheDocument()
  })
})
