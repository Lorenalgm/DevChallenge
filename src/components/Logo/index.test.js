import React from 'react';
import { render, screen } from '../../utils/test-utils';

import Logo from '.';

describe('<Logo />', () => {
    it('should render as link to home', () => {
        const { container } = render(<Logo />);

        expect(
            screen.getByRole('link', { name: /dev challenge/i })
        ).toHaveAttribute('href', '/');

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render "challenge" in yellow', () => {
        render(<Logo />);

        expect(screen.getByText('Challenge')).toHaveStyle({
            color: 'var(--yellow)',
        });
    });
});
