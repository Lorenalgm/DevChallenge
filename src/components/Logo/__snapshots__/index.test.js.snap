// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render as link to home 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: var(--white);
  cursor: pointer;
}

.c1.is-active {
  color: var(--yellow);
}

.c0 {
  font-size: 30px;
  font-weight: bold;
  color: var(--white);
}

.c0 span {
  color: var(--yellow);
}

<div
  class="c0"
>
  <a
    aria-current="page"
    class="c1 active"
    href="/"
  >
    Dev
    <span>
      Challenge
    </span>
  </a>
</div>
`;
